src/overlap_check/overlap.cpp

```diff
@@ -0,0 +1,184 @@
+
+#include "overlap.hpp"
+#include "ProgressBar.hpp"
+
+#include "moab/GeomTopoTool.hpp"
+#include "moab/GeomQueryTool.hpp"
+
+using namespace moab;
+
+ErrorCode check_location_for_overlap(std::shared_ptr<GeomQueryTool>& GQT,
+                                     const Range& all_vols,
+                                     CartVect loc,
+                                     CartVect dir,
+                                     OverlapMap& overlap_map) {
+  ErrorCode rval;
+
+  GeomTopoTool* GTT = GQT->gttool();
+  std::set<int> vols_found;
+  double bump = 1E-9;
+
+  // move the point slightly off the vertex
+  loc += dir * bump;
+
+  for (const auto& vol : all_vols) {
+    int result = 0;
+    rval = GQT->point_in_volume(vol, loc.array(), result, dir.array());
+    MB_CHK_SET_ERR(rval, "Failed point in volume for Vol with id "
+                   << GTT->global_id(vol)
+                   << " at location " << loc);
+
+    if (result == 1) {
+      vols_found.insert(GTT->global_id(vol));
+    }
+  }
+
+  if (vols_found.size() > 1) {
+    overlap_map[vols_found] = loc;
```

Comment:
Should this be the original location rather than the bumped location?  Or is there any value in a vertex ID here?

Reply:
My only concern is that if a user then tries to inspect the model at this location, they may get a different result than the location actually tested in the overlap check. Which leads us back to ideally testing exact vertex locations, but the verbosity of what comes out of MOAB when we do that is really prohibitive...

Reply:
I guess if we are using a location instead of a vertex ID, the bump doesn't matter that much - it's almost round off error.